weaviate/embedded.py

```diff
@@ -98,11 +117,15 @@ def check_supported_platform(self):
 
     def start(self):
         if self.is_listening():
-            print(f"embedded weaviate is already listing on port {self.port}")
+            print(f"embedded weaviate is already listing on port {self.options.port}")
             return
 
         self.ensure_weaviate_binary_exists()
         my_env = os.environ.copy()
+        if self.options.additional_env_vars is not None:
+            for key, value in self.options.additional_env_vars.items():
+                my_env.setdefault(key, value)
```

Comment:
is the expectation that additional_env_vars can override any defaults? in that case it should be done after line 138 and should use `my_env.update(self.options.additional_env_vars` instead.

Also I would expect people to set additional environment variables and override using native methods to set env variables. I probably prefer to remove this option and let users set environment variables instead unless there is a good use case for this?

Reply:
That's a valid use case and I can see the benefit to UX. Let's keep it in. In that case, this should move down to happen after defaults have been set and use dict.update instead.

Reply:
The issue is that setdefault should only be used to set a default: https://docs.python.org/3/library/stdtypes.html#dict.setdefault but can't be used to override any of the defaults, which I think would be what most end-users expect.

Reply:
applied your suggestion :)